### ISSUE TITLE
Add support for custom task lists

### ISSUE THREAD

**Issue by powerman** [KyleKing/mdformat-obsidian#1]
Current version breaks example from https://help.obsidian.md/Editing+and+formatting/Basic+formatting+syntax#Task+lists:

```diff
 - [x] Milk
-- [?] Eggs
-- [-] Bread
+
+- \[?\] Eggs
+
+- \[-\] Bread

```
Was:
![изображение](https://github.com/user-attachments/assets/d0f86c8b-2144-4bd9-8dcf-cb04ce393d3a)
Now:
![изображение](https://github.com/user-attachments/assets/1f67be40-e5cf-4e61-91bf-b7138ac9bfa3)

Moreover, with installed community plugin https://github.com/obsidian-tasks-group/obsidian-tasks (about 1.5M installations, so it's really widely used) task lists can use nearly any symbols, e.g.: https://publish.obsidian.md/tasks/Reference/Status+Collections/AnuPpuccin+Theme#Supported+statuses so it's better to not try to fix them at all.

**Comment by KyleKing** [comment]
Thanks for these great feature requests!

I'm traveling and have a few busy weeks, but I agree that all should be implemented and I'll try to make time when I can. I'm open to contributions, if you would like to work on any of these.

**Comment by powerman** [comment]
> I'm open to contributions, if you would like to work on any of these.

No, sorry, Python isn't my language of choice, I use Go.

**Comment by KyleKing** [comment]
@powerman, I believe I've addressed the open tickets in changes released with v0.1.0! Let me know if you run into any issues
